test/dialect/mysql/test_types.py

```diff
@@ -478,6 +479,34 @@ def test_float_type_compile(self, type_, sql_text):
         self.assert_compile(type_, sql_text)
 
 
+class TypeCompileUUIDMariaDBTest(fixtures.TestBase, AssertsCompiledSQL):
+    __only_on__ = "mariadb"
+
+    @testing.only_if("mariadb>=10.7")
+    def test_uuid(self):
+        self.assert_compile(sqltypes.Uuid(), "UUID")
+
+    @testing.only_if("mariadb<10.7")
+    def test_uuid_char(self):
+        self.assert_compile(sqltypes.Uuid(), "CHAR(32)")
+
+    @testing.combinations(
+        (sqltypes.Uuid(), (10, 6, 5), True, "CHAR(32)"),
+        (sqltypes.Uuid(), (10, 6, 5), False, "CHAR(32)"),
+        (sqltypes.Uuid(), (10, 7, 0), True, "UUID"),
+        (sqltypes.Uuid(), (10, 7, 0), False, "CHAR(32)"),
+        (sqltypes.UUID(), (10, 6, 5), True, "UUID"),
+        (sqltypes.UUID(), (10, 6, 5), False, "UUID"),
+        (sqltypes.UUID(), (10, 7, 0), True, "UUID"),
+        (sqltypes.UUID(), (10, 7, 0), False, "UUID"),
+    )
+    def test_uuid_combinations(self, type_, version, native_uuid, res):
```

Comment:
**Michael Bayer** (zzzeek) wrote:

pretty good, I might have some updates to this

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/5120

Reply:
**Michael Bayer** (zzzeek) wrote:

Done

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/5120